src/HatterJar.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.19;
+
+interface IHat {
```

Comment:
This generalizes to any 1155, so you could rename as IERC1155 if desired

Reply:
In that case could you make it 1155? Otherwise we set ourselves up for redundant/duplicate contracts